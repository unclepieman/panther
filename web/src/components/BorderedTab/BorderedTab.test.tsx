/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render, fireEvent } from 'test-utils';
import { TabList, TabPanel, TabPanels, Tabs } from 'pouncejs';
import { BorderedTab } from './index';

describe('BorderedTab', () => {
  it('renders', () => {
    const { container } = render(
      <Tabs>
        <TabList>
          <BorderedTab>1</BorderedTab>
          <BorderedTab>2</BorderedTab>
        </TabList>
        <TabPanels>
          <TabPanel>One</TabPanel>
          <TabPanel>Two</TabPanel>
        </TabPanels>
      </Tabs>
    );
    expect(container).toMatchSnapshot();
  });

  it('works like a normal `Tab` element', () => {
    const { getByText } = render(
      <Tabs>
        <TabList>
          <BorderedTab>1</BorderedTab>
          <BorderedTab>2</BorderedTab>
        </TabList>
        <TabPanels>
          <TabPanel>One</TabPanel>
          <TabPanel>Two</TabPanel>
        </TabPanels>
      </Tabs>
    );

    expect(getByText('One')).toBeInTheDocument();
    expect(getByText('Two')).not.toBeVisible();

    fireEvent.click(getByText('2'));

    expect(getByText('One')).not.toBeVisible();
    expect(getByText('Two')).toBeInTheDocument();

    fireEvent.keyDown(getByText('2'), { key: 'ArrowLeft', code: 'ArrowLeft' });

    expect(getByText('One')).toBeInTheDocument();
    expect(getByText('Two')).not.toBeVisible();
  });
});
