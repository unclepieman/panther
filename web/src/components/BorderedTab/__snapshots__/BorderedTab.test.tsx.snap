// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedTab renders 1`] = `
.panther-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-1 {
  z-index: 1;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: transparent;
  -webkit-transition: all 0.1s linear;
  transition: all 0.1s linear;
}

.panther-0 {
  font-weight: 400;
  font-size: 1.125rem;
  border-bottom: 3px solid;
  z-index: 5;
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 24px;
  padding-bottom: 24px;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-color: #0081c5;
}



.panther-2 {
  font-weight: 400;
  font-size: 1.125rem;
  border-bottom: 3px solid;
  z-index: 5;
  margin-left: 16px;
  margin-right: 16px;
  padding-top: 24px;
  padding-bottom: 24px;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-color: transparent;
}

.panther-2:hover {
  border-color: #3e516b;
}

.panther-5 {
  outline: none;
}

<div>
  <header
    id="main-header"
  />
  <div
    data-orientation="horizontal"
    data-reach-tabs=""
  >
    <div
      aria-orientation="horizontal"
      class="panther-4"
      data-reach-tab-list=""
      role="tablist"
    >
      <button
        aria-controls="tabs--1--panel--0"
        aria-selected="true"
        class="panther-1"
        data-orientation="horizontal"
        data-reach-tab=""
        data-selected=""
        id="tabs--1--tab--0"
        role="tab"
        tabindex="0"
        type="button"
      >
        <h4
          class="panther-0"
        >
          1
        </h4>
      </button>
      <button
        aria-controls="tabs--1--panel--1"
        aria-selected="false"
        class="panther-1"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--1--tab--1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <h4
          class="panther-2"
        >
          2
        </h4>
      </button>
    </div>
    <div
      data-reach-tab-panels=""
    >
      <div
        aria-labelledby="tabs--1--tab--0"
        class="panther-5"
        data-reach-tab-panel=""
        id="tabs--1--panel--0"
        role="tabpanel"
        tabindex="-1"
      >
        One
      </div>
      <div
        aria-labelledby="tabs--1--tab--1"
        class="panther-5"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--1--panel--1"
        role="tabpanel"
        tabindex="-1"
      >
        Two
      </div>
    </div>
  </div>
  <footer
    id="footer"
  />
</div>
`;
